tools/hangar/__snapshots__/test_corpus/valid/class.test.w_compile_tf-aws.md

```diff
@@ -95,8 +95,8 @@ module.exports = function({  }) {
 ```js
 module.exports = function({ $A }) {
   class B extends $A {
-    constructor(sound) {
-      super(sound);
+    async $inflight_init(sound) {
+      await super.$inflight_init(sound);
```

Comment:
What happens if super doesn't have an inflight_init function?

```suggestion
      await super?.$inflight_init?(sound);
```

(I really don't remember the syntax🤦‍♀️, but basically optional chaining to the max)

Reply:
> What happens if super doesn't have an inflight_init function?

Pretty useless calling `super()` if there's no init in the parent, but still a bug. Fixed, nice catch.

Reply:
What happens if my inflight wing class extends a normal JavaScript class? How would constructor chaining super calls work?

For example, I need to propagate arguments to the normal super ctor?

Reply:
I am not sure this is a very common scenario and maybe we decide that we don't allow it but consider this:

```js
bring "some-external-jsii-module" as e1;

inflight class MyClass extends e1.MyBase {
  init(foo: str) {
    super(foo);
    // bla
  }
}
```

In the above example, MyBase is a normal JavaScript class without an async $inflight_init.

Reply:
JSII libraries don't have inflight classes (for now?), but they can have phase-independent (i.e. non-construct) classes, and it should be possible to define an inflight class as a subclass of a phase-independent class once we have more phase-independent language capabilities, so the example sort of makes sense...

Reply:
I'll look into supporting this. If it become too complex I'll open a separate issue since I think at this point this is a fringe case, and getting inflight inits to work in all other cases is higher priority.

Reply:
Look what error I got while writing a test for this:
![image](https://github.com/winglang/wing/assets/1160578/a4a634a7-5f3b-489a-b6de-b6e19c48326e)
So I'm resolving this since it's currently not supported anyway. I'll open an issue and possibly complete the job with another PR.